src/rabbit_alarm.erl

```diff
@@ -86,9 +94,56 @@ start() ->
                           fun clear_alarm/1]),
     {ok, DiskLimit} = application:get_env(disk_free_limit),
     rabbit_sup:start_delayed_restartable_child(
-      rabbit_disk_monitor, [DiskLimit]),
+      rabbit_disk_monitor, [parse_disk_limit(DiskLimit)]),
     ok.
 
+parse_mem_limit({absolute, Limit}) ->
+    case parse_limit(Limit) of
+        {ok, ParsedLimit} -> {absolute, ParsedLimit};
+        {error, parse_error} ->
+            rabbit_log:error("Unable to parse vm_memory_high_watermark value"),
+            ?DEFAULT_VM_MEMORY_HIGH_WATERMARK
+    end;
+    {absolute, parse_limit(Limit, ?DEFAULT_VM_MEMORY_HIGH_WATERMARK)};
+parse_mem_limit(Relative) when is_float(Relative), Relative < 1 ->
+    Relative;
+parse_mem_limit(_) ->
+    ?DEFAULT_VM_MEMORY_HIGH_WATERMARK.
+
+parse_disk_limit({mem_relative, Relative}) 
+    when is_float(Relative), Relative < 1 ->
+    {mem_relative, Relative};
+parse_disk_limit(Limit) -> 
+    case parse_limit(Limit) of
+        {ok, ParsedLimit} -> ParsedLimit;
+        {error, parse_error} ->
+            rabbit_log:error("Unable to parse disk_free_limit value"),
+            ?DEFAULT_DISK_FREE_LIMIT
+    end;
+parse_disk_limit(_) -> ?DEFAULT_DISK_FREE_LIMIT.
+
+parse_limit(MemLim) when is_integer(MemLim) -> {ok, MemLim};
+parse_limit(MemLim) when is_string(MemLim) ->
```

Comment:
It would be great to have a few unit tests for this function (in rabbitmq-test).